### PR TITLE
Queue a global task to resolve  a Promise.

### PR DESCRIPTION
**Pull request by markafoltz**
Fixes Issue #43.

Promises can only be resolved (or rejected) on the main event loop.  Queue a global task to do so when the Promise returned by `restrictTo()` resolves.

***
<a href="https://pr-preview.s3.amazonaws.com/markafoltz/element-capture/pull/45.html" title="Last updated on Sep 16, 2024, 9:14 PM UTC (6106a9c)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/screen-share/element-capture/45/cb84846...markafoltz:6106a9c.html" title="Last updated on Sep 16, 2024, 9:14 PM UTC (6106a9c)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -419,11 +424,12 @@ <h3>BrowserCaptureMediaStreamTrack extension</h3>
...
                       Call the track's state before this method invocation <var>PRE-STATE</var>, and
                       after this method invocation <var>POST-STATE</var>. The user agent MUST
-                      resolve <var>p</var> when it is guaranteed that no more frames [=restricted=]
+                      <a data-cite="!HTML#queue-a-global-task">queue a global task</a> to resolve
+                      <var>p</var> when it is guaranteed that no more frames [=restricted=]
                       (or [=unrestricted=]) according to <var>PRE-STATE</var> will be delivered to
                       the application, and that any additional frames delivered to the application
-                      will therefore be [=restricted=] (or [=unrestricted=]) according to either
-                      <var>POST-STATE</var> or a later state.
+                      will therefore be [=restricted=] (or [=unrestricted=]) according to
```
Comment (eladalon1983) [inline index.html:431]: nit: I suspect my [own formatter](https://github.com/screen-share/element-capture/blob/main/.prettierrc.json) will undo this change next time I commit.

**Review by eladalon1983 — APPROVED** [review]

File: index.html (hunk 1 above)
Comment (markafoltz) [reply index.html]: Rewrapped

**Comment by markafoltz** [comment]
@eladalon1983, please merge at your convenience.

You might want to crib from other specs that run the spec through tidy-html, so the formatting stays consistent.  Example: https://github.com/w3c/presentation-api/blob/main/Makefile

**Comment by eladalon1983** [comment]
> @eladalon1983, please merge at your convenience.
> 
> You might want to crib from other specs that run the spec through tidy-html, so the formatting stays consistent. Example: https://github.com/w3c/presentation-api/blob/main/Makefile

Thanks.
I guess another step is necessary to have the Makefile trigger as a presubmit action or something? Maybe [this](https://github.com/w3c/presentation-api/blob/main/.github/workflows/auto-publish.yml) or something similar?
